File: commonutils-jre/src/main/java/net/maxsmr/commonutils/gson/exclusion/FieldExclude.java
package net.maxsmr.commonutils.gson.exclusion;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;

/** Excluding fields from Gson serialization to json via [FieldsExclusionStrategy] where transient can not be used,
 * e.g. at [android.arch.persistence.room.Entity] */
@Target(FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface FieldExclude {

}

File: commonutils-jre/src/main/kotlin/net/maxsmr/commonutils/model/gson/exclusion/FieldExclude.kt
package net.maxsmr.commonutils.model.gson.exclusion

/** Excluding fields from Gson serialization to json via [FieldsExclusionStrategy] where transient can not be used,
 * e.g. at [android.arch.persistence.room.Entity]  */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class FieldExclude